### PR TITLE
FileCache.js: switch to a Class

### PR DESCRIPTION
**Pull request by XhmikosR**
Better check the [non-whitespace diff](https://github.com/stylelint/stylelint/pull/4459/files?utf8=%E2%9C%93&diff=unified&w=1) for this. Also please confirm this is safe, I didn't test it a lot :)

### CODE REVIEW

File: lib/utils/FileCache.js [hunk 1]
```diff
@@ -13,53 +13,55 @@ const DEFAULT_HASH = '';
  * @param {string} [hashOfConfig]
  * @constructor
  */
-function FileCache(cacheLocation, hashOfConfig) {
-	const cacheFile = path.resolve(
-		getCacheFile(cacheLocation || DEFAULT_CACHE_LOCATION, process.cwd()),
-	);
+class FileCache {
+	constructor(cacheLocation, hashOfConfig) {
```
Comment (fanich37) [inline lib/utils/FileCache.js]:
```suggestion
	constructor(cacheLocation = DEFAULT_CACHE_LOCATION, hashOfConfig = DEFAULT_HASH) {
		const cacheFile = path.resolve(
			getCacheFile(cacheLocation, process.cwd()),
		);

		debug(`Cache file is created at ${cacheFile}`);
		this._fileCache = fileEntryCache.create(cacheFile);
		this._hashOfConfig = hashOfConfig;
	}
```

Comment (fanich37) [reply lib/utils/FileCache.js]: It also should fix the lint issue.

Comment (XhmikosR) [reply lib/utils/FileCache.js]: Thanks, good call :)

**Comment by fanich37** [comment]
Ooops, my bad. I thought multi-line suggestion is supported (or i did smth wrong).
I'll fix it :)

**Comment by XhmikosR** [comment]
@hudochenkov FYI you did a merge commit again

**Comment by hudochenkov** [comment]
Should I did something else instead?

**Comment by XhmikosR** [comment]
I thought you guys only did Squash and merge.

**Comment by hudochenkov** [comment]
You [expressed](https://github.com/stylelint/stylelint/pull/4462#issuecomment-569419098) fair point that all work should be credited to people who did it. So we enabled regular merges. We don't have rebase, because regular merge and squash options allow us to put PR number into commit message. This way we can track down why changes were made and all related discussions.

If all commits in PR were made by one person, then we can safely squash, because author of squashed commits would be author of original commits.

If PR has commit from more than one author, then we'll do merge so all commits are authored by person who did that commit only.

**Comment by XhmikosR** [comment]
Hmm, then I guess you changed the default merge commit message? Because right now it doesn't mention it's a merge commit hence why I was confused :)

**Comment by hudochenkov** [comment]
Yeah, I decided to change default messages to have more descriptive merge commits :)

**Pull request merged**